### PR TITLE
Stillsuit

### PR DESCRIPTION
**Pull request by Madnus**
Although it is not called such in the game itself, it is, in fact, a
stillsuit from Dune. Thirst is reduced by .3 while it is being worn.

It is based off of the phelloderm mutation for balancing purposes.
Successfully tested (roughly) for disparity between (thirst-level/time)
for wearing the suit and not wearing the suit.

It has been added to both the "vault" and "spider" item groups, with a
"5" for likelihood in both cases. Ran debug test group. Results: Vault -
4/100 and Spider - 1/200.

-M

### CODE REVIEW

File: data/json/items/armor.json [hunk 1]
```diff
@@ -4834,6 +4834,24 @@
     },
     {
         "type" : "ARMOR",
+        "id" : "stillsuit",
+        "name" : "WS-47G moisture retention suit",
+        "weight" : 3481,
+        "color" : "dark_gray",
+        "covers" : ["TORSO", "ARMS", "LEGS"],
+        "symbol" : "[",
+        "description" : "The WebbStar moisture retention suit utilizes quantum cooling technology to prevent up to 30% of moisture loss through perspiration.  The suit is powered by the wearer's movement.",
```
Comment (Darkyhard) [inline data/json/items/armor.json]: I think it's supposed to be powered in some other way, as the player can just stay still for a long time, like when sleeping while wearing this suit.

File: src/player.cpp [hunk 2]
```diff
@@ -5568,7 +5568,10 @@ void player::update_needs( int rate_multiplier )
     if( has_trait("PLANTSKIN") ) {
         thirst_rate -= 0.2f;
     }
-
+    if( is_wearing("stillsuit") ) {
```
Comment (Darkyhard) [inline src/player.cpp]: It would probably be better to make this check for an item flag than a single item.

**Comment by Malkeus** [comment]
I like this new toy. Could this be made into a flag rather than being unique to a single item? Perhaps even a flag with a modifier similar to LIGHT_[x].

**Comment by Rivet-the-Zombie** [comment]
[*Urine and feces are processed in the thigh pads.*](https://www.getyarn.io/yarn-clip/d491fac8-d843-49d3-8bfc-c1d081d23f91)

**Comment by Malkeus** [comment]
https://youtu.be/Bj7R_2WWdKs?t=26
I'd call that 10 encumbrance at the most. He could probably dance in that thing, no problem.

**Comment by kevingranade** [comment]
Cannon says it's driven by boot pumps.
Implies some additional resistance to walking.
If it's supposed to be a stillsuit, drop the "quantum" whatever, it's
supposed to be conventional technology.

**Comment by Malkeus** [comment]
There's a movement driven camouflage cloak in Bio-Weapon mod, it uses AEP_SPEED_DOWN to reduce movement speed by 20. I'm not sure a slight decrease in thirst would warrant such a large speed nerf though.

**Comment by kevingranade** [comment]
Yes, that would be excessive, but some increase is called for, 5 seems about right to me.

**Comment by Rivet-the-Zombie** [comment]
>Also, 20 encumbrance for a skin-tight suit is way too high IMO.

>I'd call that 10 encumbrance at the most.

It's got VARSIZE.
----
I like the reduced movement speed idea, and would like to second the five point movement speed reduction.

**Comment by vache** [comment]
My question is, why did we decide to make this before the cataclysm?  Is it a response to a decline in freshwater supplies?  If so, why is it so rare to spawn?  IMO, it should be a relatively common spawn inside the vault, as it could be expected to be an item issued to every inhabitant to slow their use of limited water supplies inside the vault.

**Comment by Cat09** [comment]
it could be new military tech, they have need for water conservation.

**Comment by kevingranade** [comment]
> common spawn inside the vault

The thing about a stillsuit, even in the Dune series is that you only wear it when going outside, the Siech has a built-in system for capturing ambient moisture, and I'd expect a Vault to be the same.

However having them in the Vault is still fine, presumably they would try to be prepared for all kinds of post-apocalyptic scenarios, including extreme desert environments, so having a stillsuit there makes sense.
Similarly they might appear in the prepper basement or LMOE shelter.  Military spawns might rarely have them in a "prepared for anything" kind of way as well.
I'm not sure about the spider basement, it seems like they're too rare for someone wearing one to just stumble across a bunch of spiders and get killed.

**Comment by Zireael07** [comment]
Spider basement doesn't need more ultra-rare loot.

**Pull request merged**